gz_ros2_control/src/gz_system.cpp

```diff
@@ -654,7 +654,7 @@ hardware_interface::return_type GazeboSimSystem::write(
       {
         this->dataPtr->ecm->CreateComponent(
           this->dataPtr->joints_[i].sim_joint,
-          sim::components::JointVelocityCmd({0}));
+          sim::components::JointVelocityCmd({this->dataPtr->joints_[i].joint_velocity_cmd}));
```

Comment:
This can be simplified by using the [`SetComponentData`](https://gazebosim.org/api/sim/9/classgz_1_1sim_1_1EntityComponentManager.html#ada2c5008634545bbb5d9cb3a14a3aecf) function, so it would just look like:

```c++
this->dataPtr->ecm->SetComponentData(this->dataPtr->joints_[i].sim_joint, {this->dataPtr->joints_[i].sim_joint})
```
replacing the entire if/else block starting with ` if (!this->dataPtr->ecm->Component<sim::components::JointVelocityCmd>(
          this->dataPtr->joints_[i].sim_joint))`